Week3/StringSeg.java

```diff
@@ -0,0 +1,68 @@
+import java.util.HashSet;
+import java.util.Set;
+
+public class StringSeg {
+
+	public static boolean breakable(String line, Set<String> Dictionary) {
+		
+		if(Dictionary== null && (line==null || line=="")) 
+		{
+			return true;
+		}
+		if(Dictionary == null)
+		{
+			return false;
+			
+		}	
+		if(line==null || Dictionary.isEmpty())
+		{
+			return false;
+		}
+		
+		if(line.equals("")) //nothing to search
+			return true;
+		
+	    int[] traverse = new int[line.length()+1]; //traverse entire string with one step ahead
+	    
+	    for(int i=0;i<traverse.length;i++)
+	    {
+	    	traverse[i]=-1;							//all are unvisited.
+	    }

```

Comment:
You seem to be using this as a binary visited/not visited, bool would be a better choice for this.

Reply:
this also has the benefit that bool[] is automatically initialized to false for all of the elements, so you can skip this initialization